choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerDBService.java

```diff
@@ -164,6 +195,65 @@ public ChoreographerStep addNextStepsToStep(final String stepName, final Set<Str
+                            stepDetail.setVersion(service.getVersionRequirement());
+                            stepDetail.setType(type);
+                            stepDetail.setDto(ow.writeValueAsString(ofrRequestDTO));
+                            stepDetail.setStep(step);
+                        }
+
+                        if (service.getMinVersionRequirement() != null && service.getMaxVersionRequirement() != null && service.getVersionRequirement() == null) {
+                            if (service.getMinVersionRequirement() < service.getMaxVersionRequirement()) {
+                                stepDetail.setServiceDefinition(service.getServiceDefinitionRequirement());
+                                stepDetail.setMaxVersion(service.getMaxVersionRequirement());
+                                stepDetail.setMinVersion(service.getMinVersionRequirement());
+                                stepDetail.setType(type);
+                                stepDetail.setDto(ow.writeValueAsString(ofrRequestDTO));
+                                stepDetail.setStep(step);
+                            } else {
+                                throw new InvalidParameterException("Minimum version must be lesser than maximum version.");
+                            }
+                        }
+
+                        if (service.getVersionRequirement() != null && service.getMinVersionRequirement() != null && service.getMaxVersionRequirement() != null) {
+                            throw new InvalidParameterException("A service can only have one version or one min/max version pair defined.");
+                        }
+                    } else {
+                        throw new InvalidParameterException("The service definition description of the requested service is null or empty.");
+                    }
+                } else {
+                    throw new InvalidParameterException("The requested service in the OFR is null.");
+                }
+            } else {
+                throw new InvalidParameterException("The OFR request for the step is null.");
+            }
+        } catch (JsonProcessingException ex) {
+            logger.debug(ex.getMessage(), ex);
+        } catch (InvalidParameterException ex) {
+            throw ex;
+        } catch (ArrowheadException ex) {
+            logger.debug(ex.getMessage(), ex);
+            throw new ArrowheadException(CoreCommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
+        }
+
+        return choreographerStepDetailRepository.saveAndFlush(stepDetail);
+    }
+
+
     //-------------------------------------------------------------------------------------------------
 	@Transactional(rollbackFor = ArrowheadException.class)
     public ChoreographerAction createAction(final String name, final List<String> firstStepNames, final long planId, final List<ChoreographerStepRequestDTO> steps) {
```

Comment:
should be private 
planId => plan entity
name and content of firstStepNames should be normalized